### ISSUE TITLE
Ungraded 3 level 3

### ISSUE THREAD

**Issue by Delocy** [nus-cs2030/2223-s2#839]
I dont get why my current implementation does work. Any ideas why? 
<img width="710" alt="Screenshot 2023-04-09 at 5 19 59 PM" src="https://user-images.githubusercontent.com/94375191/230764945-5e41809f-6fff-4b61-8766-1fac4b0714fd.png">

**Comment by granolaluvr69** [comment]
> I dont get why my current implementation does'nt work. Any ideas why? 
> 
> <img width="710" alt="Screenshot 2023-04-09 at 5 19 59 PM" src="https://user-images.githubusercontent.com/94375191/230764945-5e41809f-6fff-4b61-8766-1fac4b0714fd.png">
> 
> 

any errors or screenshots of expr not working as intended?

**Comment by Delocy** [comment]
> > I dont get why my current implementation does'nt work. Any ideas why?
> > <img alt="Screenshot 2023-04-09 at 5 19 59 PM" width="710" src="https://user-images.githubusercontent.com/94375191/230764945-5e41809f-6fff-4b61-8766-1fac4b0714fd.png">
> 
> any errors or screenshots of expr not working as intended?
<img width="358" alt="Screenshot 2023-04-09 at 5 50 52 PM" src="https://user-images.githubusercontent.com/94375191/230766119-5f26cf3a-7bb1-4459-a695-5e460b5ca8db.png">
im getting 20 instead of 14 for this

**Comment by granolaluvr69** [comment]
in your second if condition, you can use the operator stored in the left hand side to operate on the values in the left hand side expression.

p.s. optional .get() methods aren't allowed, might wanna reconsider your approach. you are allowed to use orelseget, orelse and orelsethrow though.

**Comment by wadeyman** [comment]
is it a must to use Optionals?

**Comment by granolaluvr69** [comment]
Nope, but a stream implementation would probably require us to construct our own implementation of a pair class, which you probably wouldn't have the time to do in an exam scenario.
